sw/airborne/subsystems/ins/ins_vectornav.h

```diff
@@ -0,0 +1,186 @@
+ * any later version.
+ *
+ * paparazzi is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with paparazzi; see the file COPYING.  If not, write to
+ * the Free Software Foundation, 59 Temple Place - Suite 330,
+ * Boston, MA 02111-1307, USA.
+ */
+/**
+ * @file ins_vectornav.h
+ *
+ * Vectornav VN-200 INS subsystem
+ *
+ * @author Michal Podhradsky <michal.podhradsky@aggiemail.usu.edu>
+ */
+#ifndef INS_VECTORNAV_H
+#define INS_VECTORNAV_H
+
+// Subsystems
+#include "subsystems/imu.h"
+#include "subsystems/gps.h"
+#include "subsystems/ahrs.h"
+#include "subsystems/ins.h"
+
+// Peripheral
+#include "peripherals/vn200_serial.h"
+
+// Geodetic / Math
+#include "math/pprz_algebra.h"
+#include "math/pprz_geodetic_int.h"
+#include "math/pprz_algebra_float.h"
+#include "math/pprz_geodetic_int.h"
+#include "math/pprz_isa.h"
+
+// Generated
+#include "generated/airframe.h"
+#include "generated/flight_plan.h"
+
+// Systime
+#include "mcu_periph/sys_time.h"
+
+// Abi
+#include "subsystems/abi.h"
+
+#if GPS_USE_LATLONG
+/* currently needed to get nav_utm_zone0 */
+#include "subsystems/navigation/common_nav.h"
+#include "math/pprz_geodetic_float.h"
+#endif
+
+/*
+ * IMU Subsystem defines
+ */
+extern void imu_vectornav_event(void);
+
+#if !USE_NPS
+#define ImuEvent imu_vectornav_event
+#endif /* Only for non-NPS targets */
+
+/*
+ * IMU scalling defaults - still needed, although not used
+ * Filled in with dummy values (ones)
+ */
+#if !defined IMU_GYRO_P_SENS & !defined IMU_GYRO_Q_SENS & !defined IMU_GYRO_R_SENS
+#define IMU_GYRO_P_SENS 1
+#define IMU_GYRO_P_SENS_NUM 1
+#define IMU_GYRO_P_SENS_DEN 1
+#define IMU_GYRO_Q_SENS 1
+#define IMU_GYRO_Q_SENS_NUM 1
+#define IMU_GYRO_Q_SENS_DEN 1
+#define IMU_GYRO_R_SENS 1
+#define IMU_GYRO_R_SENS_NUM 1
+#define IMU_GYRO_R_SENS_DEN 1
+#endif
+
+#if !defined IMU_ACCEL_X_SENS & !defined IMU_ACCEL_Y_SENS & !defined IMU_ACCEL_Z_SENS
+#define IMU_ACCEL_X_SENS 1
+#define IMU_ACCEL_X_SENS_NUM 1
+#define IMU_ACCEL_X_SENS_DEN 1
+#define IMU_ACCEL_Y_SENS 1
+#define IMU_ACCEL_Y_SENS_NUM 1
+#define IMU_ACCEL_Y_SENS_DEN 1
+#define IMU_ACCEL_Z_SENS 1
+#define IMU_ACCEL_Z_SENS_NUM 1
+#define IMU_ACCEL_Z_SENS_DEN 1
+#endif
+
+
+/*
+ * GPS subsystem defines
+ */
+#if !USE_NPS
+#define GpsEvent(_x) ;
+#endif /* Only for non-NPS targets */
+
+/*
+ * AHRS subsystem
+ * - uses ins_vectonav_wrapper.h
+ */
+
+/*
+ * INS subsystem defines
+ */
+#ifndef DefaultInsImpl
+#define DefaultInsImpl ins_vectornav
+#endif
+
+
+
+
+// Ins implementation state (fixed point)
+struct InsVectornav {
+  struct LtpDef_i  ltp_def; // initial position
+  bool_t           ltp_initialized; // status indicator
+
+  // output LTP NED for telemetry messages
+  struct NedCoor_i ltp_pos;
```

Comment:
I don't see any particular reason to have these in the InsVectornav struct the way you use those right now...
IMHO it would make more sense to just grab those from the state interface (like you do already anyway) when actually needed in the `send_ins` function. That should even save some conversions...

Ideally you would also not grab those from the state interface, but directly convert the float values from the vectornav INS...
Since hopefully we will have proper support for multiple INS soon and the one sent to the state interface might not be the one from vectornav....
Unfortunately we don't have a component id (or something similar) in the messages right now, so this is still a problem if multiple components send the same message... but better prepare properly for this right now as it doesn't cost much...